app/copy_tree.py

```diff
@@ -93,6 +93,11 @@ def _copy_preserve(self, src: Path, dst_dir: Path):
             self._copy_stat(src, dst_path)
         # src is plain file or symlink
         elif src.is_file() or src.is_symlink():  # includes broken symlink
+            if src.is_symlink() and dst_path.exists():
+                # When source is symlink, shutil.copy2 fails to overwrite destination file.
+                # To avoid this, remove destination file beforehand.
+                os.remove(str(dst_path))
```

Comment:
こんな感じでどうでしょうか？
```suggestion
            if src.is_symlink() or dst_path.is_symlink() or dst_path.is_file():
                # When source is symlink, shutil.copy2 fails to overwrite destination file.
                # To avoid this, remove destination file beforehand.
                dst_path.unlink()
            elif dst_path.is_dir():
                raise OtaErrorRecoverable(f"{dst_path} exists as a directory")
```

Reply:
`src file -> dst symlink: shutil.copy2 success but unintended behavior. src is copied to dst link target.`

でもこの場合にうまくcopyできないので、 src がfileの場合でもdstを削除するケースがあると思います！

Reply:
今回、CIで検出したものと逆のパターンですね。
確かに、src が plain fileの場合も dstは削除した方がいいですね。

Reply:
srcがplain file でも symlinkでもdstのファイルが存在していたら、削除すればいいので以下のようにしました。

- dst が plain fileの時: dstを削除
- dst が symlinkの時: dstを削除
- dst が directoryの時: exception